pkg/controller/nodeipam/legacyprovider.go

```diff
@@ -62,10 +63,10 @@ func createLegacyIPAM(
 	}
 	ipamc, err := ipam.NewController(cfg, kubeClient, cloud, cidr, serviceCIDR, nodeCIDRMaskSizes[0])
 	if err != nil {
-		klog.Fatalf("Error creating ipam controller: %v", err)
+		return nil, errors.Wrapf(err, "Error creating ipam controller")
```

Comment:
I think that the new conventions is to use `fmt.Errorf("Error creating ipam controller: %w", err)`

```
Wrapping errors with %w
```
https://go.dev/blog/go1.13-errors

Reply:
Updated